cocos/core/game.ts

```diff
@@ -968,67 +972,50 @@ export class Game extends EventTarget {
         }
 
         this.on(Game.EVENT_HIDE, () => {
-            legacyCC.game.pause();
+            this.pause();
         });
         this.on(Game.EVENT_SHOW, () => {
-            legacyCC.game.resume();
+            this.resume();
         });
     }
 
-    private _loadRenderPipeline (restart: Function | null) {
-        const useSplash = (!EDITOR && !PREVIEW && legacyCC.internal.SplashScreen);
-
-        // Load render pipeline if needed
-        const renderPipeline = this.config.renderPipeline;
-        if (renderPipeline) {
-            legacyCC.loader.load({ uuid: renderPipeline }, (err, asset) => {
-                // failed load renderPipeline
-                if (err || !(asset instanceof RenderPipeline)) {
-                    console.warn(`Failed load renderpipeline: ${renderPipeline}, engine failed to initialize, will fallback to default pipeline`);
-                    console.warn(err);
-                    this._setRenderPipeline();
-                }
-                else {
-                    try {
-                        this._setRenderPipeline(asset);
-                    } catch (e) {
-                        console.warn(e);
-                        console.warn(`Failed load renderpipeline: ${renderPipeline}, engine failed to initialize, will fallback to default pipeline`);
-                        this._setRenderPipeline();
-                    }
-                }
-                this._safeEmit(Game.EVENT_GAME_INITED);
-                if (useSplash) {
-                    const splashScreen = legacyCC.internal.SplashScreen.instance as SplashScreen;
-                    splashScreen.main(legacyCC.director.root);
-                    splashScreen.setOnFinish(() => {
-                        if (this.onStart) { this.onStart(); }
-                        if (restart) { restart(); }
-                    });
-                    splashScreen.loadFinish = true;
-                }
-                else {
-                    if (this.onStart) { this.onStart(); }
-                    if (restart) { restart(); }
-                }
-            });
-        }
-        else {
-            this._setRenderPipeline();
+    private _setupRenderPipelineAndShowSplashScreen () {
+        return Promise.resolve(this._setupRenderPipeline()).then(() => {
             this._safeEmit(Game.EVENT_GAME_INITED);
-            if (useSplash) {
-                const splashScreen = legacyCC.internal.SplashScreen.instance as SplashScreen;
-                splashScreen.main(legacyCC.director.root);
-                splashScreen.setOnFinish(() => {
-                    if (this.onStart) { this.onStart(); }
-                    if (restart) { restart(); }
+            if (this.onStart) {
+                this.onStart();
+            }
+            return this._showSplashScreen();
+        });
```

Comment:
onStart should be executed after splashScreen finish.

Reply:
Updated.